src/main/java/com/direwolf20/buildinggadgets/common/tools/ExchangingModes.java

```diff
@@ -1,208 +1,125 @@
-                    if (isReplaceable(world, pos, currentBlock, setBlock, fuzzyMode)) {
-                        coordinates.add(pos);
-                    }
-                }
-            }
-        }
-        //***************************************************
-        //TorchPlacer
-        //***************************************************
-        else if (mode == GadgetExchanger.ToolMode.Grid) {
-            range++;
-            for (int x = range * -7 / 5; x <= range * 7 / 5; x++) {
-                for (int z = range * -7 / 5; z <= range * 7 / 5; z++) {
-                    if (x % (((range - 2) % 6) + 2) == 0 && z % (((range - 2) % 6) + 2) == 0) {
-                        pos = new BlockPos(startBlock.getX() + x, startBlock.getY(), startBlock.getZ() + z);
-                        if (isReplaceable(world, pos, currentBlock, setBlock, fuzzyMode)) {
-                            coordinates.add(pos);
-                        }
-                    }
-                }
-            }
-        }
-        return coordinates;
+    public static List<BlockPos> collectPlacementPos(World world, EntityPlayer player, BlockPos hit, EnumFacing sideHit, ItemStack tool, BlockPos initial) {
+        IBuildingMode mode = byName(NBTTool.getOrNewTag(tool).getString("mode")).getModeImplementation();
+        return mode.createExecutionContext(player, hit, sideHit, tool).collectFilteredSequence(world, tool, player, initial);
     }
+
+    public static BiPredicate<BlockPos, IBlockState> combineTester(World world, ItemStack tool, EntityPlayer player, BlockPos initial) {
+        IBlockState initialBlockState = world.getBlockState(initial);
+        IBlockState target = GadgetUtils.getToolBlock(tool);
+        return (pos, state) -> {
+            IBlockState worldBlockState = world.getBlockState(pos);
+
+            // Don't try to replace for the same block
+            if (worldBlockState == target)
+                return false;
+
+            // No need to replace if source and target are the same if fuzzy mode is off
+            if (!GadgetGeneric.getFuzzy(tool) && worldBlockState != initialBlockState)
+                return false;
+
+            // If the target is already enqueued, don't replace it
+            if (worldBlockState == ModBlocks.effectBlock.getDefaultState())
+                return false;
+            // Only replace existing blocks, don't place more
+            if (worldBlockState.getBlock().isAir(worldBlockState, world, pos))
+                return false;
+
+            TileEntity tile = world.getTileEntity(pos);
+            // Only replace construction block with same block state
+            if (tile instanceof ConstructionBlockTileEntity)
+                if (((ConstructionBlockTileEntity) tile).getBlockState() == state)
+                    return false;
+
+            // Otherwise if the block has a tile entity, ignore it
+            if (tile != null)
```

Comment:
```suggestion
            else if (tile != null)
```
Your comment sugggests this to be an else (Otherwise...). Not sure whether this should be an else, you need to take a look whether the comment is correct or the code is correct.